### PR TITLE
resource/configmap: add

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/5461.

This is mostly generated by the script below.
The only manual changes are in this commit: https://github.com/giantswarm/operatorkit/pull/288/commits/a4ae7bccb6056e943e1d62601d4133243d4b42a0.

```bash
replace() {
    local a=$1; shift
    local b=$1; shift
    local dir=$1; shift
    # escape backshashes
    local a="${a//\//\\/}"
    local b="${b//\//\\/}"
    for file in $(find $dir \( -path ./.git -o -path ./vendor \)  -prune -o -type f -print); do
        sed -i -e "s/${a}/${b}/g" $file
    done
}
cp ../release-operator/pkg/controller/resource/app ./resource/configmap
replace 'package app' 'package configmap' ./resource/configmap
replace 'App CR' ConfigMap ./resource/configmap
replace 'AppCR' ConfigMap ./resource/configmap
replace 'appCR' configMap ./resource/configmap
replace 'v1alpha1.App' 'v1.ConfigMap' ./resource/configmap
replace '"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"' '"k8s.io/api/core/v1"' ./resource/configmap
replace 'g8sClient.ApplicationV1alpha1().Apps' 'k8sClient.CoreV1().ConfigMaps' ./resource/configmap
goimports -w ./resource/configmap
```

### CODE REVIEW

File: resource/configmap/create.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package configmap
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	v1 "k8s.io/api/core/v1"
```
Comment (rossf7) [inline resource/configmap/create.go]:
```suggestion
	corev1 "k8s.io/api/core/v1"
```

Minor but I prefer this alias.

File: resource/configmap/spec.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+type StateGetter interface {
+	// GetCurrentState returns a current state of the system for the given
+	// carnation of the observed Kubernetes object.
```
Comment (rossf7) [inline resource/configmap/spec.go]:
```suggestion
	// incarnation of the observed Kubernetes object.
```

Pretty sure is incarnation. Nice use of the word!

Maybe its stating the obvious but sometimes we want to create a single configmap other times its multiple. Should that be mentioned in the godoc? Could be useful for future people integrating the resource.

Comment (kopiczko) [reply resource/configmap/spec.go]: IDK. This is fine to produce single CM as a current/desired state. I can add a sentence here.

File: resource/configmap/create.go (hunk 1 above)
Comment (kopiczko) [reply resource/configmap/create.go]: Done. `corev1` import is used now.

File: resource/configmap/spec.go (hunk 2 above)
Comment (kopiczko) [reply resource/configmap/spec.go]: Added this
```
	// incarnation of the observed Kubernetes object. The state consists of
	// multiple objects but it is fine to return a slice of single object
	// if only one object is managed by this resource.
```

**Comment by kopiczko** [comment]
This is tested in https://github.com/giantswarm/app-operator/pull/51 by @tomahawk28. Going.

**Pull request merged**